### PR TITLE
GLT-1091 make service records CRUDable by all

### PR DESCRIPTION
**Pull request by callunity**

### CODE REVIEW

**Comment by apmasell** [comment]
The code is good, but policy question: do we want to allow non-internal users to edit?

**Comment by callunity** [comment]
The only non-internal users are techs and external users, right? Techs should be able to add them (at least, they had this capacity prior to this commit) and external users can only see the external webapp, which is read-only. Service Records also aren't available to edit via a REST API, so I don't believe it'll be an issue.

**Comment by apmasell** [comment]
Okay. :+1:

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/UserAuthMisoRequestManager.java [hunk 1]
```diff
@@ -1778,10 +1778,8 @@ public void deleteSequencerReference(SequencerReference sequencerReference) thro
   }
 
   @Override
-  public void deleteSequencerServiceRecord(uk.ac.bbsrc.tgac.miso.core.data.SequencerServiceRecord serviceRecord) throws IOException {
-    if (getCurrentUser().isAdmin()) {
-      backingManager.deleteSequencerServiceRecord(serviceRecord);
-    }
+  public void deleteSequencerServiceRecord(SequencerServiceRecord serviceRecord) throws IOException {
+    backingManager.deleteSequencerServiceRecord(serviceRecord);
```
Comment (djcooke) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/UserAuthMisoRequestManager.java]: Should all users be able to delete?

**Comment by djcooke** [comment]
I think delete should be admin only, but :+1: if I'm wrong

**Comment by callunity** [comment]
I can get behind admin-only delete, though that means that the sequencing team would be emailing us to do it. Any objections?

**Comment by callunity** [comment]
(For what it's worth, non-admins are able to edit and delete records in the current LIMS)

**Comment by djcooke** [comment]
Having to delete a service record should be pretty rare. I assume that lab managers will be admins in MISO as well though?

**Comment by callunity** [comment]
@tonydebat was suggesting that maybe lab managers won't need to be admins in MISO, since there aren't that many lab-facing admin things. And right now the only admins in GSLE are Lee, Danielle, and Faridah.

**Comment by djcooke** [comment]
The question is whether all users should be able to delete service records. If yes, merge as is. If no, there are several solutions:

1. lab asks us to delete them in the rare case that it's needed
2. make lab managers admin so they can handle it instead
3. make another security group for managers that has extra privileges, but not as much as admin. Let that group delete the records. Either
   * do this now (not likely)
   * use one of the other options for now and do this later

I don't have a strong opinion on what is done. Just being wary of "everyone can delete"

**Pull request merged**